### PR TITLE
Filter teams by name, org members by email

### PR DESCRIPTION
**Pull request by sebasslash**
Two new query params have been added to the list endpoints for `Teams` and `OrganizationMemberships`:

`filter[names]` - List Teams
`filter[email]` - List Org Members

### CODE REVIEW

File: validations.go [hunk 1]
```diff
@@ -20,3 +21,9 @@ func validString(v *string) bool {
 func validStringID(v *string) bool {
 	return v != nil && reStringID.MatchString(*v)
 }
+
+// validEmail checks if the given input is a correct email
+func validEmail(v string) bool {
+	_, err := mail.ParseAddress(v)
```
Comment (Uk1288) [inline validations.go:27]: nice 🎉

File: organization_membership.go [hunk 2]
```diff
@@ -77,6 +77,9 @@ type OrganizationMembershipListOptions struct {
 	// Optional: A list of relations to include. See available resources
 	// https://www.terraform.io/cloud-docs/api-docs/organization-memberships#available-related-resources
 	Include []OrgMembershipIncludeOpt `url:"include,omitempty"`
+
+	// Optional: A list of organization member emails to filter by.
+	Emails []string `url:"filter[email],omitempty"`
```
Comment (Uk1288) [inline organization_membership.go:82]: should this be, 
```suggestion
	Emails []string `url:"filter[emails],omitempty"`
```

Comment (sebasslash) [reply organization_membership.go:82]: 👋 The param in the API is actually `filter[email]`: https://www.terraform.io/cloud-docs/api-docs/organization-memberships#query-parameters

File: team.go [hunk 3]
```diff
@@ -278,3 +285,13 @@ func validateTeamIncludeParams(params []TeamIncludeOpt) error {
 
 	return nil
 }
+
+func validateTeamNames(names []string) error {
+	for _, name := range names {
```
Comment (brandonc) [inline team.go:290]: This works when names is nil, right?

**Review by brandonc — APPROVED** [review]
✅

File: team.go (hunk 3 above)
Comment (sebasslash) [reply team.go:290]: Yessir!

**Review by Uk1288 — APPROVED** [review]
LGTM 🎉

File: errors.go [hunk 4]
```diff
@@ -262,4 +262,8 @@ var (
 	ErrRequiredWorkspacesList = errors.New("no workspaces list provided")
 
 	ErrCommentBody = errors.New("comment body is required")
+
+	ErrEmptyTeamName = errors.New("team name can not be empty")
```
Comment (annawinkler) [inline errors.go:266]:
```suggestion
	ErrEmptyTeamName = errors.New("team name cannot be empty")
```

**Review by annawinkler — APPROVED** [review]
⭐ Nice!

**Pull request merged**